migrations/20240723090305_publish_view_comment.sql

```diff
@@ -0,0 +1,19 @@
+-- stores the comments on a published view
+CREATE TABLE IF NOT EXISTS af_published_view_comment (
+  comment_id          UUID NOT NULL DEFAULT gen_random_uuid(),
+  -- comments are never deleted, only marked as deleted, unless we intentionally wants to clean
+  -- the tables by removing the comments from the database
+  reply_comment_id    UUID REFERENCES af_published_view_comment(comment_id) ON DELETE CASCADE,
+  -- The view id should exists on af_published_collab, However, we can't enforce this foreign key
+  -- constraint because af_published_collab primary key is (workspace_id, view_id)
```

Comment:
ideally we should delete the comments when the it's unpublished, not sure if adding `workspace_id` column can help with this, or might have to add a postgres trigger to achieve. if not deleted, then if user publish a doc, comments, unpublished and publish the same doc again, the comments is probably still there

Reply:
According to the requirement, if it is the same view id, the comments should still be preserved, similar to how a blog post page still keep the comments despite update.

Reply:
That said, the lack of constraint is bothering me as well - because that means it is possible for comment to be posted on non existent view id.